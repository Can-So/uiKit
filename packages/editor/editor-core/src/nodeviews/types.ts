export interface ReactNodeProps {
  selected: boolean;
}
export type ReactComponentConstructor = new () => React.Component<any, any>;
export type ProsemirrorGetPosHandler = () => number;
