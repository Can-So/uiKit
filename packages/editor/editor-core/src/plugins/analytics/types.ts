import { Dispatch } from '../../event-dispatcher';

type AEP<Action, ActionSubject, ActionSubjectID, Attributes, EventType> = {
  action: Action;
  actionSubject: ActionSubject;
  actionSubjectId?: ActionSubjectID;
  attributes?: Attributes;
  eventType: EventType;
};

type UIAEP<Action, ActionSubject, ActionSubjectID, Attributes> = AEP<
  Action,
  ActionSubject,
  ActionSubjectID,
  Attributes,
  'ui'
>;

type TrackAEP<Action, ActionSubject, ActionSubjectID, Attributes> = AEP<
  Action,
  ActionSubject,
  ActionSubjectID,
  Attributes,
  'track'
>;

/* UI events */

type ButtonAEP<ActionSubjectID, Attributes> = UIAEP<
  'clicked',
  'button',
  ActionSubjectID,
  Attributes
>;

type PickerAEP<ActionSubjectID, Attributes> = UIAEP<
  'opened',
  'picker',
  ActionSubjectID,
  Attributes
>;

type TypeAheadAEP<ActionSubjectID, Attributes> = UIAEP<
  'invoked',
  'typeAhead',
  ActionSubjectID,
  Attributes
>;

type EditorStartAEP = UIAEP<
  'started',
  'editor',
  undefined,
  { platform: 'mobileNative' | 'mobileHybrid' | 'web' }
>;

type EditorStopAEP = UIAEP<
  'stopped',
  'editor',
  'save' | 'cancel',
  {
    inputMethod: 'toolbar' | 'shortcut';
    documentSize: number;
    nodeCount?: {
      tables: number;
      headings: number;
      lists: number;
      mediaSingles: number;
      mediaGroups: number;
      panels: number;
      extensions: number;
      decisions: number;
      actions: number;
      codeBlocks: number;
    };
  }
>;

type ButtonHelpAEP = ButtonAEP<
  'helpButton',
  { inputMethod: 'shortcut' | 'toolbar' }
>;

type ButtonFeedbackAEP = ButtonAEP<'feedbackButton', undefined>;

type PickerEmojiAEP = PickerAEP<'emojiPicker', { inputMethod: 'toolbar' }>;

type PickerImageAEP = PickerAEP<
  'cloudPicker',
  { inputMethod: 'toolbar' | 'quickInsert' }
>;

type TypeAheadQuickInsertAEP = TypeAheadAEP<
  'quickInsertTypeAhead',
  { inputMethod: 'keyboard' }
>;

type TypeAheadEmojiAEP = TypeAheadAEP<
  'emojiTypeAhead',
  { inputMethod: 'quickInsert' | 'keyboard' }
>;

type TypeAheadLinkAEP = TypeAheadAEP<
  'linkTypeAhead',
  {
    inputMethod: 'toolbar' | 'quickInsert' | 'shortcut';
  }
>;

type TypeAheadMentionAEP = TypeAheadAEP<
  'mentionTypeAhead',
  {
    inputMethod: 'toolbar' | 'quickInsert' | 'keyboard';
  }
>;

type UIEventPayload =
  | EditorStartAEP
  | EditorStopAEP
  | ButtonHelpAEP
  | ButtonFeedbackAEP
  | PickerEmojiAEP
  | PickerImageAEP
  | TypeAheadQuickInsertAEP
  | TypeAheadEmojiAEP
  | TypeAheadLinkAEP
  | TypeAheadMentionAEP;

/** Text input and formatting events */

type FormatAEP<ActionSubjectID, Attributes> = TrackAEP<
  'formatted',
  'text',
  ActionSubjectID,
  Attributes
>;

type SubstituteAEP<ActionSubjectID, Attributes> = TrackAEP<
  'autoSubstituted',
  'text',
  ActionSubjectID,
  Attributes
>;

type FormatBasicAEP = FormatAEP<
  'strong' | 'italic' | 'underline' | 'code' | 'strike',
  {
    inputMethod: 'toolbar' | 'shortcut' | 'autoformatting' | 'floatingToolbar';
  }
>;

type FormatSuperSubAEP = FormatAEP<
  'superscript' | 'subscript',
  {
    inputMethod: 'toolbar';
  }
>;

type FormatIndentationAEP = FormatAEP<
  'indentation',
  {
    inputMethod: 'toolbar' | 'keyboard';
    direction: 'indent' | 'outdent';
    previousIndentationLevel: number;
    newIndentLevel: number;
    indentType: 'paragraph' | 'list' | 'heading' | 'codeBlock';
  }
>;

type FormatHeadingAEP = FormatAEP<
  'heading',
  {
    inputMethod: 'toolbar' | 'shortcut' | 'autoformatting';
    newHeadlineLevel: 0 | 1 | 2 | 3 | 4 | 5 | 6;
  }
>;

type FormatBlockQuoteAEP = FormatAEP<
  'blockQuote',
  {
    inputMethod: 'toolbar' | 'keyboard' | 'autoformatting' | 'quickInsert';
  }
>;

type FormatClearAEP = FormatAEP<
  'clearFormatting',
  {
    inputMethod: 'toolbar' | 'shortcut';
    formattingCleared: string[];
  }
>;

type FormatColorAEP = FormatAEP<
  'color',
  {
    newColor: string;
    previousColor: string;
  }
>;

type FormatListAEP = FormatAEP<
  'numberedList' | 'bulletedList',
  {
    inputMethod: 'toolbar' | 'keyboard' | 'autoformatting' | 'quickInsert';
  }
>;

type SubstituteProductAEP = SubstituteAEP<
  'productName',
  {
    product: string;
    originalSpelling: string;
  }
>;

type SubstituteSymbolAEP = SubstituteAEP<
  'symbol',
  {
    symbol: 'rightArrow' | 'leftArrow';
  }
>;

type SubstitutePuncAEP = SubstituteAEP<
  'punctuation',
  {
    punctuation: 'emDash' | 'ellipsis' | 'singleQuote' | 'doubleQuote';
  }
>;

type FormatSubstitutionEventPayload =
  | FormatBasicAEP
  | FormatSuperSubAEP
  | FormatIndentationAEP
  | FormatHeadingAEP
  | FormatBlockQuoteAEP
  | FormatClearAEP
  | FormatColorAEP
  | FormatListAEP
  | SubstituteProductAEP
  | SubstituteSymbolAEP
  | SubstitutePuncAEP;

/** Insertion events */

type InsertAEP<ActionSubjectID, Attributes> = TrackAEP<
  'inserted',
  'document',
  ActionSubjectID,
  Attributes
>;

type InsertDividerAEP = InsertAEP<
  'divider',
  {
    inputMethod: 'quickInsert' | 'toolbar' | 'autoformatting';
  }
>;

type InsertPanelAEP = InsertAEP<
  'panel',
  { inputMethod: 'quickInsert' | 'toolbar' }
>;

type InsertCodeBlockAEP = InsertAEP<
  'codeBlock',
  {
    inputMethod: 'quickInsert' | 'toolbar' | 'autoformatting' | 'insertMenu';
  }
>;

type InsertEventPayload =
  | InsertDividerAEP
  | InsertPanelAEP
  | InsertCodeBlockAEP;

export type AnalyticsEventPayload =
  | UIEventPayload
  | FormatSubstitutionEventPayload
  | InsertEventPayload;

export type AnalyticsDispatch = Dispatch<{
  payload: AnalyticsEventPayload;
  channel?: string;
}>;
