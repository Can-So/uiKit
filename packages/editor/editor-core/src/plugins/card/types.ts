export type CardAppearance = 'inline' | 'block';
export type CardType = 'smart-card' | 'custom' | 'unsupported';

export interface CardProvider {
  getType(url: string): CardType;
  getData(url: string, appearance: CardAppearance): Promise<any>;
}

export interface CardOptions {
  provider?: Promise<CardProvider>;
}
