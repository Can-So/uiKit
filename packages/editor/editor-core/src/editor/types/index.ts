import { EditorState, Transaction } from 'prosemirror-state';

export * from './editor-instance';
export * from './editor-config';
export * from './editor-plugin';
export * from './editor-props';
export * from './editor-appearance-component';

export type Command = (
  state: EditorState,
  dispatch: (tr: Transaction) => void,
) => boolean;
