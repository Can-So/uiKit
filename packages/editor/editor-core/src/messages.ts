// Common Translations will live here
import { defineMessages } from 'react-intl';

export default defineMessages({
  layoutFixedWidth: {
    id: 'fabric.editor.layoutFixedWidth',
    defaultMessage: 'Center',
    description:
      'Display your element (image, table, extension, etc) as standard width',
  },
  layoutWide: {
    id: 'fabric.editor.layoutWide',
    defaultMessage: 'Wide',
    description:
      'Display your element (image, table, extension, etc) wider than normal',
  },
  layoutFullWidth: {
    id: 'fabric.editor.layoutFullWidth',
    defaultMessage: 'Full width',
    description:
      'Display your element (image, table, extension, etc) as full width',
  },
  remove: {
    id: 'fabric.editor.remove',
    defaultMessage: 'Remove',
    description:
      'Delete the element (image, panel, table, etc.) from your document',
  },
  visit: {
    id: 'fabric.editor.visit',
    defaultMessage: 'Open link in a new window',
    description: '',
  },
});
