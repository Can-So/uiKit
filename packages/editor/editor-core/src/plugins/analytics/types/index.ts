export * from './enums';
export * from './events';
export * from './format-events';
export * from './substitute-events';
export * from './ui-events';
export * from './insert-events';
