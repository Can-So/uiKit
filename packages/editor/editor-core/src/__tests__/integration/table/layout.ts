import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';

import {
  editable,
  getDocFromElement,
  fullpage,
  forEach,
  changeSelectedNodeLayout,
  animationFrame,
  toggleBreakout,
} from '../_helpers';

import {
  defaultTableInOverflow,
  defaultTableResizedTable,
  nestedTables,
} from './__fixtures__/layout-documents';

import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';

import { TableCssClassName } from '../../../plugins/table/types';

import messages from '../../../messages';

BrowserTestCase(
  'Remains in overflow on table scale to wide',
  { skip: ['ie', 'edge', 'safari', 'firefox'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await page.browser.windowHandleMaximize();

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(defaultTableInOverflow),
      allowTables: {
        advanced: true,
      },
    });

    await page.waitForSelector(TableCssClassName.TOP_LEFT_CELL);
    await page.click(TableCssClassName.TOP_LEFT_CELL);

    // Change layout to wide
    await page.waitForSelector(`.${TableCssClassName.LAYOUT_BUTTON}`);
    await page.click(`.${TableCssClassName.LAYOUT_BUTTON}`);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Remains in overflow on table scale to full width',
  { skip: ['ie', 'edge', 'safari', 'firefox'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await page.browser.windowHandleMaximize();

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(defaultTableInOverflow),
      allowTables: {
        advanced: true,
      },
    });

    await page.waitForSelector(TableCssClassName.TOP_LEFT_CELL);
    await page.click(TableCssClassName.TOP_LEFT_CELL);

    // Toggle layout twice.
    await forEach([0, 1], async _ => {
      await page.waitForSelector(`.${TableCssClassName.LAYOUT_BUTTON}`);
      await page.click(`.${TableCssClassName.LAYOUT_BUTTON}`);
    });

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Maintains the wide layout size without overflow',
  { skip: ['ie', 'edge', 'safari', 'firefox'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await page.browser.windowHandleMaximize();

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(defaultTableResizedTable),
      allowTables: {
        advanced: true,
      },
    });

    await page.waitForSelector(TableCssClassName.TOP_LEFT_CELL);
    await page.click(TableCssClassName.TOP_LEFT_CELL);

    await page.waitForSelector(`.${TableCssClassName.LAYOUT_BUTTON}`);
    await page.click(`.${TableCssClassName.LAYOUT_BUTTON}`);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Maintains the wide layout size without overflow with dynamic text sizing',
  { skip: ['ie', 'edge', 'safari', 'firefox'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(defaultTableResizedTable),
      allowTables: {
        advanced: true,
      },
      allowDynamicTextSizing: true,
    });

    await page.waitForSelector(TableCssClassName.TOP_LEFT_CELL);
    await page.click(TableCssClassName.TOP_LEFT_CELL);

    await page.waitForSelector(`.${TableCssClassName.LAYOUT_BUTTON}`);
    await page.click(`.${TableCssClassName.LAYOUT_BUTTON}`);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Maintains the full-width layout size without overflow',
  { skip: ['ie', 'edge', 'safari', 'firefox'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await page.browser.windowHandleMaximize();

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(defaultTableResizedTable),
      allowTables: {
        advanced: true,
      },
    });

    await page.waitForSelector(TableCssClassName.TOP_LEFT_CELL);
    await page.click(TableCssClassName.TOP_LEFT_CELL);

    await forEach([0, 1], async _ => {
      await page.waitForSelector(`.${TableCssClassName.LAYOUT_BUTTON}`);
      await page.click(`.${TableCssClassName.LAYOUT_BUTTON}`);
    });

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Maintains the default layout size without overflow when toggling through layouts',
  { skip: ['ie', 'edge', 'safari', 'firefox'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await page.browser.windowHandleMaximize();

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(defaultTableResizedTable),
      allowTables: {
        advanced: true,
      },
    });

    await page.waitForSelector(TableCssClassName.TOP_LEFT_CELL);
    await page.click(TableCssClassName.TOP_LEFT_CELL);

    await forEach([0, 1, 2], async _ => {
      await page.waitForSelector(`.${TableCssClassName.LAYOUT_BUTTON}`);
      await page.click(`.${TableCssClassName.LAYOUT_BUTTON}`);
    });

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Scales down column sizes when bodied extension parent layout changes',
  { skip: ['ie', 'edge', 'firefox', 'safari'] },
  async client => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(nestedTables),
      allowTables: {
        advanced: true,
      },
      allowExtension: {
        allowBreakout: true,
      },
      allowLayouts: {
        allowBreakout: true,
      },
      allowBreakout: true,
    });

    await page.waitForSelector('.extension-container p');
    await page.click('.extension-container p');
    await changeSelectedNodeLayout(
      page,
      messages.layoutFixedWidth.defaultMessage,
    );

    await animationFrame(page);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Scales down column sizes when parent layout changes breakout',
  { skip: ['ie', 'edge', 'firefox', 'safari'] },
  async client => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(nestedTables),
      allowTables: {
        advanced: true,
      },
      allowExtension: {
        allowBreakout: true,
      },
      allowLayouts: {
        allowBreakout: true,
      },
      allowBreakout: true,
    });

    await page.waitForSelector('div[data-layout-section]');
    await page.click('div[data-layout-section]');
    await toggleBreakout(page, 2);

    await animationFrame(page);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
