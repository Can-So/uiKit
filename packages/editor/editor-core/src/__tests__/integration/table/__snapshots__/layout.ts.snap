// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maintains the default layout size without overflow when toggling through layouts 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  340,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  340,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  340,
                ],
                "rowspan": 2,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  340,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  340,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  340,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  340,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Maintains the full-width layout size without overflow 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "full-width",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  900,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  900,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  900,
                ],
                "rowspan": 2,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  900,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  900,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  900,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  900,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Maintains the wide layout size without overflow 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "wide",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  480,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  480,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  480,
                ],
                "rowspan": 2,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  480,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  480,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  480,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  480,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Maintains the wide layout size without overflow with dynamic text sizing 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "wide",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  481,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  479,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  481,
                ],
                "rowspan": 2,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  479,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  479,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  481,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  479,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Remains in overflow on table scale to full width 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "full-width",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  595,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  1223,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  131,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  595,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  1223,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  131,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  595,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  1223,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  131,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`Remains in overflow on table scale to wide 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "wide",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  318,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  653,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  69,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  318,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  653,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  69,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  318,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  653,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  69,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;
